src/Get-VcxprojNugetPackageVersions.ps1

```diff
@@ -0,0 +1,339 @@
+#Requires -Version 5.1
+
+<#
+.SYNOPSIS
+    Recursively searches for .vcxproj files and extracts NuGet package versions from associated packages.config files.
+
+.DESCRIPTION
+    This script searches directories below the script location for Visual Studio C++ project files (.vcxproj)
+    and their corresponding packages.config files. It extracts all NuGet package references with their versions
+    and generates a comprehensive report.
+
+.PARAMETER OutputFormat
+    Specifies the output format: Table, CSV, JSON, or Object. Default is Table. Object format returns PackageInfo objects without printing anything.
+
+.PARAMETER ExportPath
+    Optional path to export the results to a file. Not applicable for Object format.
+
+.PARAMETER PackageFilter
+    Optional filter to search for specific package names. Supports wildcards. If specified, only packages matching this filter will be included in the results.
+
+.EXAMPLE
+    .\Get-VcxprojNugetPackageVersions.ps1
+    
+.EXAMPLE
+    .\Get-VcxprojNugetPackageVersions.ps1 -OutputFormat CSV -ExportPath "packages-report.csv"
+
+.EXAMPLE
+    .\Get-VcxprojNugetPackageVersions.ps1 -PackageFilter "Microsoft*"
+
+.EXAMPLE
+    .\Get-VcxprojNugetPackageVersions.ps1 -PackageFilter "Newtonsoft.Json" -OutputFormat JSON
+
+.EXAMPLE
+    $packages = .\Get-VcxprojNugetPackageVersions.ps1 -OutputFormat Object
+#>
+
+[CmdletBinding()]
+param(
+    [ValidateSet("Table", "CSV", "JSON", "Object")]
+    [string]$OutputFormat = "Table",
+    
+    [string]$ExportPath,

```

Comment:
Do we need this? If I want to write to file I can just `| Out-File` (and apply whatever format I want inbetween)